keps/sig-instrumentation/1668-trace-popagating/README.md

```diff
@@ -0,0 +1,501 @@
+
+below is a key/value pair example in object annotation :
+
+| key               | value(encoded)                       | origin value(decoded)                                   | description                                                  |
+| ---------------------------- | ------------------------------------------------------- | ------------------------------------------------------------ | ------------------------------------------------------------ |
+| trace.kubernetes.io/context | 0/3kO3imLJzu3N54RTuOEUDbc2C0poIMAA== | 00-ca057eae1a26b66314fe3e361eedc5ca-3696483da6bfdcea-00 |it consists of `<version>-<traceid>-<spanid>-<flag>`. <br />A corresponding http header field is like `traceparent: 00-ca057eae1a26b66314fe3e361eedc5ca-3696483da6bfdcea-00` which is a w3c specific [trace-context](https://w3c.github.io/trace-context/#traceparent-header ). <br />|
+
+### Behaviors with and without Mutating webhook
+Since the mutating webhook is optional for users, we will explain the different behaviors between with and without mutating webhook.
+
+#### with Mutating webhook
+
+**kubectl request**:
+
+- APIServer uses otel to start a new Span
+- APIServer uses otel to propagate `span context` to the other end(webhook)
+- Webhook persists `span context` to object
+
+**controllers request:**
+
+- Controller uses otel to start a related Span, which connected to the `span context` in object
+- Controller uses otel to propagate `span context`  to the other end(APIServer)
+- APIServer uses otel  to start a related Span, which connected to the `span context` from the  incoming request
+- APIServer uses otel to propagate `span context`  to the other end(webhook)
+- Webhook persists `span context` to object
+
+#### without Mutating webhook
+
+**kubectl request:**
+
+- APIServer uses otel to start a new Span
+- APIServer uses otel to propagate `span context` to the other end
+- ~~Webhook persists `span context` to object~~
+
+**controllers request:**
+
+- Controller start uses otel to start a new Span
+- Controller uses otel to propagate `span context`  to the other end(APIServer)
+- APIServer uses otel  to start related Span, which connected to the `span context` from the  incoming request
+- APIServer uses otel to propagate `span context` to the other end
+- ~~Webhook persists `span context`  to object~~
+
+In short, the webhook decides whether to add `span context` to the object.
+
+### Test Plan
+
+All added code will be covered by unit tests.
+
+### Graduation Criteria
+
+#### Alpha
+
+- Feature covers 3 important workload objects: Deployment, Statefulset, Daemonset
+- Related unit tests described in this KEP are completed
```

Comment:
I think we should add (non-conformance) e2es for alpha as well if we can.

Reply:
I see, thanks.